implementations/rust/ockam/ockam_abac/src/policies/storage.rs

```diff
@@ -0,0 +1,146 @@
+use crate::policies::ResourceTypePolicy;
+use crate::{
+    Action, Env, Expr, PolicyAccessControl, ResourceName, ResourcePoliciesRepository,
+    ResourcePolicy, ResourceType, ResourceTypePoliciesRepository,
+};
+use ockam_core::compat::sync::Arc;
+use ockam_core::compat::vec::Vec;
+use ockam_identity::{Identifier, IdentitiesAttributes};
+use tracing::{debug, instrument};
+
+#[derive(Clone)]
+pub struct Policies {
+    resources_repository: Arc<dyn ResourcePoliciesRepository>,
+    resource_types_repository: Arc<dyn ResourceTypePoliciesRepository>,
+}
+
+impl Policies {
+    pub fn new(
+        resources_repository: Arc<dyn ResourcePoliciesRepository>,
+        resource_types_repository: Arc<dyn ResourceTypePoliciesRepository>,
+    ) -> Self {
+        Self {
+            resources_repository,
+            resource_types_repository,
+        }
+    }
+
+    #[instrument(skip_all, fields(resource_name = %resource_name, action = %action, env = %env, authority = %authority))]
+    pub async fn make_policy_access_control(
+        &self,
+        identities_attributes: Arc<IdentitiesAttributes>,
+        resource_name: ResourceName,
+        action: Action,
+        env: Env,
+        authority: Identifier,
+    ) -> ockam_core::Result<PolicyAccessControl> {
+        debug!(
+            "set a policy access control for resource '{}' and action '{}'",
+            &resource_name, &action
+        );
+        Ok(PolicyAccessControl::new(
+            self.clone(),
+            identities_attributes,
+            authority,
+            env,
+            resource_name,
+            action.clone(),
+        ))
+    }
+}
+
+// Methods for resource policies
+impl Policies {
+    pub async fn set_policy_for_resource_name(
+        &self,
+        resource_name: &ResourceName,
+        resource_type: Option<&ResourceType>,
+        action: &Action,
+        expression: &Expr,
+    ) -> ockam_core::Result<()> {
+        self.resources_repository
+            .set_policy(resource_name, resource_type, action, expression)
+            .await
+    }
+
+    pub async fn get_policy_for_resource_name(
+        &self,
+        resource_name: &ResourceName,
+        action: &Action,
+    ) -> ockam_core::Result<Option<ResourcePolicy>> {
+        self.resources_repository
+            .get_policy(resource_name, action)
+            .await
+    }
+
+    pub async fn get_policies_for_resource_name(
```

Comment:
How is that different from `get_policy_for_resource_name`? Does it return policies for all possible actions?

Reply:
> Does it return policies for all possible actions?

Exactly, this is the method we use from the `policy list` command. I think it's more useful this way so that you don't have to go action by action to see all the policies you have for a resource.